anvil/src/misc.rs

```diff
@@ -0,0 +1,19 @@
+use smithay::wayland_server::protocol::wl_data_device_manager::DndAction;
+
+pub fn dnd_action_chooser(available: DndAction, preferred: DndAction) -> DndAction {
```

Comment:
Is there a reason to do anything different, then this function is doing?
Maybe we can provide this function as a default directly in smithay?

Reply:
Yes, the wayland protocol files gives examples such as a compositor changing its behavior depending on key presses (possibly overriding the client preference). For example stuff like "holding `shift` for copy and holding `control` for move".

We could provide this function in smithay for users that don't want to customize it, though.